src/views/SystemProfile/index.tsx

```diff
@@ -0,0 +1,307 @@
+import React, { useEffect, useLayoutEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link, NavLink, useParams } from 'react-router-dom';
+import {
+  Breadcrumb,
+  BreadcrumbBar,
+  BreadcrumbLink,
+  Grid,
+  PrimaryNav,
+  SideNav,
+  SummaryBox
+} from '@trussworks/react-uswds';
+import classnames from 'classnames';
+
+import BookmarkCardIcon from 'components/BookmarkCard/BookmarkCardIcon';
+import Footer from 'components/Footer';
+import Header from 'components/Header';
+import UswdsReactLink from 'components/LinkWrapper';
+import MainContent from 'components/MainContent';
+import PageHeading from 'components/PageHeading';
+import PageWrapper from 'components/PageWrapper';
+import CollapsableLink from 'components/shared/CollapsableLink';
+import {
+  DescriptionDefinition,
+  DescriptionTerm
+} from 'components/shared/DescriptionGroup';
+import SectionWrapper from 'components/shared/SectionWrapper';
+import useCheckResponsiveScreen from 'utils/checkMobile';
+import NotFound from 'views/NotFound';
+import { mockSystemInfo } from 'views/Sandbox/mockSystemData';
+
+import sideNavItems from './components/index';
+
+import './index.scss';
+
+const SystemProfile = () => {
+  const { t } = useTranslation('systemProfile');
+  const isMobile = useCheckResponsiveScreen('tablet');
+  const [isMobileSideNavExpanded, setIsMobileSideNavExpanded] = useState(false);
+  const mobileSideNav = useRef<any>();
```

Comment:
Some notes on typing this. The `useRef()` call can be typed as `useRef<HTMLDivElement | null>(null)`. This would allow `handleClick()` to be condensed a bit:
```
const handleClick = (e: Event) => {
    if (mobileSideNav.current?.contains(e.target as HTMLElement)) {
      return;
    }

    setIsMobileSideNavExpanded(false);
  };
```
This does require a typecast on `e.target`, though. I think that's fairly safe, but it's up to you.

Reply:
That more succinct.  This was a code snippet from an existing piece of code in the app.  This is better!

Reply:
You can also change the `useRef()` call on this line to `useRef<HTMLDivElement | null>(null)`, which constrains the type a bit rather than having an `any` type floating around.